### PR TITLE
wallet: Use single FastRandomContext when creating a wallet tx

### PR DESCRIPTION
**Pull request by maflcko**
Passing around a single randomness context shouldn't come with any downsides, but documents better where randomness is used and allows the unit test to be deterministic, if they wish to be so.

### CODE REVIEW

**Review by glozow — COMMENTED** [review]
Concept ACK

File: src/wallet/coinselection.cpp [hunk 1]
```diff
@@ -191,16 +191,14 @@ std::optional<SelectionResult> SelectCoinsSRD(const std::vector<OutputGroup>& ut
     return std::nullopt;
 }
 
-static void ApproximateBestSubset(const std::vector<OutputGroup>& groups, const CAmount& nTotalLower, const CAmount& nTargetValue,
+static void ApproximateBestSubset(FastRandomContext& insecure_rand, const std::vector<OutputGroup>& groups, const CAmount& nTotalLower, const CAmount& nTargetValue,
```
Comment (promag) [inline src/wallet/coinselection.cpp:194]: nit, why reason to be the 1st arg?

**Review by promag — COMMENTED** [review]
Code review ACK fa7deaa0464576a229b5a6ab13ad033c16d0dada.

File: src/wallet/coinselection.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/wallet/coinselection.cpp:194]: It can't be the last arg, because that requires a default value, so I picked the "least-last" arg (aka first)

**Review by glozow — COMMENTED** [review]
light code review ACK fa7deaa0464576a229b5a6ab13ad033c16d0dada

**Comment by achow101** [comment]
ACK fa7deaa0464576a229b5a6ab13ad033c16d0dada

**Pull request merged**